src/OnboardingSPA/components/ThemeContextProvider/index.js

```diff
@@ -1,28 +1,54 @@
 import { useState, useEffect, createContext } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as nfdOnboardingStore } from '../../store';
+import { setFlow } from '../../utils/api/flow';
 import { THEME_DARK, THEME_LIGHT } from '../../../constants';
 
 const ThemeContext = createContext();
 
-function getPreferredColorScheme() {
+function usePreferredColorScheme() {
+	const { currentData } = useSelect( ( select ) => {
+		return {
+			currentData:
+				select( nfdOnboardingStore ).getCurrentOnboardingData(),
+		};
+	} );
+
+	if ( currentData.sitegenThemeMode ) {
+		return currentData.sitegenThemeMode;
+	}
+
 	if (
 		window.matchMedia &&
 		window.matchMedia( '(prefers-color-scheme: dark)' ).matches
 	) {
 		return THEME_DARK;
 	}
+
 	return THEME_LIGHT;
 }
 
 const ThemeProvider = ( { children } ) => {
-	const [ theme, setTheme ] = useState( getPreferredColorScheme );
+	const preferredColorScheme = usePreferredColorScheme();
```

Comment:
Considering here we have to call **usePreferredColorScheme** does it make sense to initialize `currentData` here and as well as in the above func or send as an param to the func